packages/core/utils/src/dml/entity.ts

```diff
@@ -7,5 +13,44 @@ import { RelationshipType, SchemaType } from "./types"
 export class DmlEntity<
   Schema extends Record<string, SchemaType<any> | RelationshipType<any>>
 > {
+  #cascades: EntityCascades<string[]> = {}
   constructor(public name: string, public schema: Schema) {}
+
+  /**
+   * Parse entity to get its underlying information
+   */
+  parse(): {
+    name: string
+    schema: SchemaType<any> | RelationshipType<any>
+    cascades: EntityCascades<string[]>
+  } {
+    return {
+      name: this.name,
+      schema: this.schema as unknown as SchemaType<any> | RelationshipType<any>,
+      cascades: this.#cascades,
+    }
+  }
+
+  /**
+   * Delete actions to be performed when the entity is deleted. For example:
+   *
+   * You can configure relationship data to be deleted when the current
+   * entity is deleted.
+   */
+  cascades(
+    options: EntityCascades<
+      ExtractEntityRelations<Schema, "hasOne" | "hasMany">
+    >
+  ) {
+    options.delete?.forEach((relationship) => {
+      if (this.schema[relationship] instanceof BelongsTo) {
+        throw new Error(
+          `Cannot cascade delete "${relationship}" relationship from "${this.name}" entity. Child to parent cascades are not allowed`
+        )
+      }
+    })
```

Comment:
```suggestion
const wrongDeleteCascadeRelation = options.delete?.filter((relationship) => {
  return this.schema[relationship] instanceof BelongsTo
})

if (!!wrongDeleteCascadeRelation.length) {
  throw new Error(
    `Cannot cascade delete "${wrongDeleteCascadeRelation.join(
      ", "
    )}" relationship from "${this.name}" entity. Child to parent cascades are not allowed`
  )
}
```

The idea behind this suggestion being that you don't have to run after fixing the first issue and then get the next one, giving the opportunity to the user to fix all of them on the entity at once but also can be easily improve forfuture configuration with little effort. wdyt?

Reply:
Makes sense!